src/lib/field-next/_core.layout.scss

```diff
@@ -0,0 +1,297 @@
+  padding-block-end: calc($floating-offset + $resize-container-input-padding-block-end-base);
+}
+
+@mixin resize-container-slotted-floating-input {
+  padding-block-start: calc($floating-offset + ($resize-container-input-offset-base)) !important; // TODO: find a way to remove this !important
+  padding-block-end: #{$resize-container-input-padding-block-end-base} !important;
+}
+
+@mixin resize-container-child {
+  align-items: center;
+  block-size: #{token('container-block-size', custom)};
+}
+
+@mixin resize-container-accessory-slot {
+  @include resize-container-child;
+
+  display: flex;
+  align-self: start;
+}
+
+//
+// Container
+//
+
+@mixin container-layout {
+  display: grid;
+  position: relative;
+
+  flex-grow: 1;
+
+  grid-template-areas: 'start center end popover-icon accessory';
+  grid-template-columns: auto 1fr auto auto auto;
+}
+
+@mixin last-container-element {
+  padding-inline-end: #{token('container-padding-end')};
+}
+
+@mixin main-content-padding-inline($side, $next-sibling) {
+  @if $side == 'start' and $next-sibling == 'start' {
+    @include override('main-content-padding-start', token('start-padding-end'), value);
+  } @else if $side == 'end' {
+    @if $next-sibling == 'end' {
+      @include override('main-content-padding-end', token('end-padding-start'), value);
+    } @else if $next-sibling == 'popover-icon' {
+      @include override('main-content-padding-end', token('popover-icon-padding-start'), value);
+    } @else if $next-sibling == 'accessory' {
+      @include override('main-content-padding-end', token('accessory-padding-start'), value);
+    }
+  }
+}
+
+//
+// Floating label
+//
+
+// The floating label and input are offset away from each other by half the larger of the two line
+// heights. Additional space is added or subtracted to account for the block size of the container
+// -- one pixel for every 8 pixels above or below 56 pixels.
+$floating-offset-base: #{calc(max(typography.variable(body2, line-height), typography.variable(label2, line-height)) / 2)};
+$floating-offset-adjustment: #{calc((token('container-block-size', custom) - 56px) / 8)};
+$floating-offset: #{calc($floating-offset-base + $floating-offset-adjustment)};
+
+@mixin floating-label {
+  @include override('floating-label-translation', 0 calc($floating-offset * -1), value);
+  @include typography.style(label2);
+  @include floating-animation(floating-label-animation);
+
+  translate: #{token(floating-label-translation, custom)};
+}
+
+@mixin floating-input {
+  @include floating-animation(floating-input-animation);
+
+  box-sizing: border-box;
+}
+
+@mixin slotted-floating-input {
+  padding-block-start: calc($floating-offset * 2) !important; // TODO: find a way to remove this !important
+}
+
+@mixin floating-animation($animation-name) {
+  animation-name: #{$animation-name};
+  animation-duration: #{token(inset-animation-duration)};
+  animation-timing-function: #{token(inset-animation-timing-function)};
+}
```

Comment:
I'd be curious what it looks like to animate the label when unfloating. It may appear a bit more smooth after clearing text. If this isn't trivial to add we can add a task for looking into this later. I only suggest this because our old floating label implementation does animate back to its unfloated/resting position when blurring the field.